akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -91,28 +101,59 @@ private[stream] object Timers {
       })
 
       final override protected def onTimer(key: Any): Unit =
-        if (nextDeadline.isOverdue())
+        if (nextDeadline < System.nanoTime)
           failStage(new TimeoutException(s"No elements passed in the last $timeout."))
 
-      override def preStart(): Unit = schedulePeriodically("IdleTimeoutCheckTimer", interval = idleTimeoutCheckInterval(timeout))
+      override def preStart(): Unit = schedulePeriodically(IdleTimeoutCheckTimer, idleTimeoutCheckInterval(timeout))
     }
 
     override def toString = "IdleTimeout"
+
+    case object IdleTimeoutCheckTimer
+  }
+
+  final class BackpressureTimeout[T](timeout: FiniteDuration) extends SimpleLinearGraphStage[T] {
+    override def initialAttributes = DefaultAttributes.backpressureTimeout
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new TimerGraphStageLogic(shape) {
```

Comment:
If you convert to
```scala
... = new TimerGraphStageLogic(shape) with InHandler with OutHandler {
  override def onPush(): Unit = {...}
  override def onPull(): Unit = {...}
}
```

you can call `setHandlers(in, out, this)` below.

That would work for most of the other stages, too. ;)